src/routers/Gateway.sol

```diff
@@ -71,17 +59,19 @@ contract ConnectorGateway {
     event Pause();
     event Unpause();
 
-    constructor(address connector_, address router_, address pauseAdmin_, address delayedAdmin_, uint256 shortScheduleWait_, uint256 longScheduleWait_, uint256 gracePeriod_) {
+    constructor(address connector_, address router_, address pauseAdmin_, address delayedAdmin, uint256 shortScheduleWait_, uint256 longScheduleWait_, uint256 gracePeriod_) {
         connector = ConnectorLike(connector_);
         router = RouterLike(router_);
         SHORT_SCHEDULE_WAIT = shortScheduleWait_;
         LONG_SCHEDULE_WAIT = longScheduleWait_;
         GRACE_PERIOD = gracePeriod_;
-        pauseAdmin = AdminLike(pauseAdmin_);
-        delayedAdmin = DelayedAdminLike(delayedAdmin_);
 
         wards[msg.sender] = 1;
         emit Rely(msg.sender);
+        wards[pauseAdmin_] = 1;
+        emit Rely(pauseAdmin_);
+        wards[delayedAdmin] = 1;
+        emit Rely(delayedAdmin);
```

Comment:
I think we don't need to pass these on the constructor but can rather just `rely` these after deploying the gateway from the deployment script.